fhir-remote-index/src/main/java/com/ibm/fhir/remote/index/api/BatchParameterValue.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.remote.index.api;
+
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.remote.index.database.ParameterNameValue;
+
+/**
+ * A parameter value batched for later processing
+ */
+public abstract class BatchParameterValue {
+    protected final String requestShard;
+    protected final ParameterNameValue parameterNameValue;
+    protected final String resourceType;
+    protected final String logicalId;
+    protected final long logicalResourceId;
+
+    /**
+     * Protected constructor
+     * @param requestShard
+     * @param resourceType
+     * @param logicalId
```

Comment:
remind me what this one is again?  the (external) value of the Resource.id element of the resource?
do we need that here if we always know the (internal) logicalResourceId?

Reply:
logicalId is the Resource.id. It's useful to carry this in the object in case we hit an error. When we collect batches of parameters, they may refer to more than one logical resource, and so we may want to log the type/logicalId instead of the logical_resource_id because it's more meaningful (which assumes, of course that our logs will contain PHI).

Reply:
having javadoc (in all the places) might be useful for folks like me, so that we remember:
* logicalId is the "id" element of the resource
* resourceId is the "row" identifier for a particular version of the resource
* logicalResourceId is the "row" identifier for the logical resource

its too late to update the database, but personally I wouldn't mind adding the word "row" to all these variables within the JDBC impl.  Then it would become:
* logicalId
* resourceRowId
* logicalResourceRowId